### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by vabalas**
**This is a draft PR.** Following the helpful feedback from @kozlovic (See #400) I want to wrap the culprit `freeaddrinfo()` with a null check. 

Sorry for disturbing you because of a minor nuisance, but could you please indicate in which header should I put it (since `util.h` is not included everywhere)?

I appreciate your help!

### CODE REVIEW

File: src/util.c [hunk 1]
```diff
@@ -1373,6 +1373,13 @@ nats_ReadFile(natsBuffer **buffer, int initBufSize, const char *fn)
     return NATS_UPDATE_ERR_STACK(s);
 }
 
+void nats_FreeAddrInfo(struct addrinfo *res)
```
Comment (kozlovic) [inline src/util.c:1376]: To follow the code style and fix the issue reported on compile, I would write like this:
```
void
nats_FreeAddrInfo(struct addrinfo *res)
{
    // Calling freeaddrinfo(NULL) is undefined behaviour.
    if (res == NULL)
        return;

    freeaddrinfo(res);
}
```

File: src/comsock.c [hunk 2]
```diff
@@ -156,8 +156,7 @@ natsSock_ConnectTcp(natsSockCtx *ctx, const char *phost, int port)
     if (numIPs == 0)
     {
         for (i=0; i<numServInfo; i++)
-            if (servInfos[i])
-                freeaddrinfo(servInfos[i]);
+            nats_FreeAddrInfo(servInfos[i]);
```
Comment (kozlovic) [inline src/comsock.c:159]: You need to add:
```
#include "util.h"
```
at top of this file.

**Review by kozlovic — CHANGES_REQUESTED** [review]
Suggested some changes to fix compilation errors.

**Comment by vabalas** [comment]
I appreciate your help very much! I added the amendments you requested.

**Review by kozlovic — APPROVED** [review]
LGTM

**Pull request merged**